### PR TITLE
docs(map, tree): Explicitly call out in SharedMap and SharedTree map node APIs that no enumeration ordering is guaranteed

### PR DESCRIPTION
**Pull request by Josmithr**
In both of these APIs, we attempt to mirror the JavaScript Map APIs, which guarantee enumeration ordering matches insertion order. Neither SharedMap nor SharedTree make such guarantees. This PR adds documentation to make this clear to the end-user.

[AB#7955](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7955)

### CODE REVIEW

File: packages/dds/map/src/interfaces.ts [hunk 1]
```diff
@@ -334,12 +334,17 @@ export interface ISharedMapEvents extends ISharedObjectEvents {
...
+ * The SharedMap distributed data structure can be used to store key-value pairs.
+ *
+ * @remarks
+ * SharedMap provides the same API for setting and retrieving values that JavaScript developers are accustomed to with the
  * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map | Map} built-in object.
  * However, the keys of a SharedMap must be strings, and the values must either be a JSON-serializable object or a
  * {@link @fluidframework/datastore#FluidObjectHandle}.
  *
+ * Note: unlike JavaScript maps, SharedMap does not make any guarantees regarding enumeration order.
+ * If your scenario requires consistent key ordering, you will need to manage that ordering yourself.
```
Comment (noencke) [inline packages/dds/map/src/interfaces.ts:346]: nit: it's probably not necessary to include this sentence, given how strong the above sentence is, and it's also not clear what it means (ordering myself, like sort? or is there some other way to order the keys of the sharedtree?). The docs on the iterators themselves give a suggestion ("sort the entries")

File: packages/dds/tree/src/feature-libraries/flex-tree/flexTreeTypes.ts [hunk 2]
```diff
@@ -329,6 +329,8 @@ export interface FlexTreeMapNode<in out TSchema extends FlexMapNodeSchema> exten
 	 *
 	 * @remarks
 	 * All fields under a map implicitly exist, but `keys` will yield only the keys of fields which contain one or more nodes.
+	 *
+	 * No guarantees are made regarding the order of the entries returned.
```
Comment (noencke) [inline packages/dds/tree/src/feature-libraries/flex-tree/flexTreeTypes.ts:333]:
```suggestion
	 * No guarantees are made regarding the order of the keys returned.
```

File: packages/dds/tree/src/feature-libraries/flex-tree/flexTreeTypes.ts [hunk 3]
```diff
@@ -337,6 +339,8 @@ export interface FlexTreeMapNode<in out TSchema extends FlexMapNodeSchema> exten
 	 *
 	 * @remarks
 	 * All fields under a map implicitly exist, but `values` will yield only the fields containing one or more nodes.
+	 *
+	 * No guarantees are made regarding the order of the entries returned.
```
Comment (noencke) [inline packages/dds/tree/src/feature-libraries/flex-tree/flexTreeTypes.ts:343]:
```suggestion
	 * No guarantees are made regarding the order of the values returned.
```

**Review by noencke — APPROVED** [review]

File: packages/dds/map/src/interfaces.ts (hunk 1 above)
Comment (Josmithr) [reply packages/dds/map/src/interfaces.ts:346]: Fair. I'll just delete the second sentence then.
```suggestion
```

**Pull request merged**